### PR TITLE
[DOCS] Reformat rank feature query

### PR DESCRIPTION
**Pull request by jrodewig**
Updates the `rank_feature` query to use the new query format.

This creates separate sections for the example requests, parameters, and notes.

This is part of #40977, an effort to standardize documentation for query types.

### Preview
http://elasticsearch_44975.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/query-dsl-rank-feature-query.html

http://elasticsearch_44975.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/query-filter-context.html

### CODE REVIEW

File: docs/reference/query-dsl/rank-feature-query.asciidoc [hunk 1]
```diff
@@ -120,31 +159,83 @@ GET test/_search
...
+
+[[rank-feature-query-notes]]
+==== Notes
+
+[[rank-feature-query-saturation]]
+===== Saturation
+The `saturation` function gives a score equal to `S / (S + pivot)`, where `S` is
+the value of the rank feature field and `pivot` is a configurable pivot value so
+that the result will be less than `0.5` if `S` is less than pivot and greater
+than `0.5` otherwise. Scores are always between `0` and `1`.
```
Comment (jtibshirani) [inline docs/reference/query-dsl/rank-feature-query.asciidoc]: Perhaps we could keep the notation `(0, 1)` or say "Scores are always between `0` and `1` (exclusive)"?

File: docs/reference/query-dsl/rank-feature-query.asciidoc [hunk 2]
```diff
@@ -203,23 +291,21 @@ GET test/_search
...
 // CONSOLE
-// TEST[continued]
-
-[float]
-==== Sigmoid
 
-This function is an extension of `saturation` which adds a configurable
+[[rank-feature-query-sigmoid]]
+===== Sigmoid
+The `sigmoid` function is an extension of `saturation` which adds a configurable
```
Comment (jtibshirani) [inline docs/reference/query-dsl/rank-feature-query.asciidoc]: Same thought here about the interval notation.

File: docs/reference/query-dsl/rank-feature-query.asciidoc [hunk 3]
```diff
@@ -120,31 +159,83 @@ GET test/_search
...
+`saturation`::
++
+--
+(Optional, <<rank-feature-query-saturation,function object>>) Saturation
+function used to boost <<query-filter-context,relevance scores>> based on the
+value of the rank feature `field`. If no function is provided, the `rank_feature`
+query defaults to the `saturation` function. See
+<<rank-feature-query-saturation,Saturation>> for more information.
+
+You cannot use the `saturation` parameter with the `log` or `sigmoid`
```
Comment (jtibshirani) [inline docs/reference/query-dsl/rank-feature-query.asciidoc]: Small comment, perhaps it'd be clearer to say "Only one function `saturation`, `log`, or `sigmoid` can be provided".

**Review by jtibshirani — COMMENTED** [review]
It is really nice to see the new format! I left some minor comments.

File: docs/reference/query-dsl/rank-feature-query.asciidoc [hunk 4]
```diff
@@ -120,31 +159,83 @@ GET test/_search
...
+
+`field`::
+(Required, string) <<rank-feature,`rank_feature`>> or
+<<rank-features,`rank_features`>> field used to boost
+<<query-filter-context,relevance scores>>.
+
+`boost`::
++
+--
+(Optional, float) Floating point number used to decrease or increase
```
Comment (jtibshirani) [inline docs/reference/query-dsl/rank-feature-query.asciidoc]: One other note: I noticed that [other query documentation](https://www.elastic.co/guide/en/elasticsearch/reference/7.2/query-dsl-dis-max-query.html) places the information in parentheses after the description.

Comment (jrodewig) [reply docs/reference/query-dsl/rank-feature-query.asciidoc]: I have an open PR to update that other query documentation: #44703

The format is based on the [API Reference template](https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc), which changed.

File: docs/reference/query-dsl/rank-feature-query.asciidoc (hunk 3 above)
Comment (jrodewig) [reply docs/reference/query-dsl/rank-feature-query.asciidoc]: Thanks for the suggestion. I changed this sentence with 22352b0.

File: docs/reference/query-dsl/rank-feature-query.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/query-dsl/rank-feature-query.asciidoc]: Re-added with 890b6b3.

File: docs/reference/query-dsl/rank-feature-query.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/query-dsl/rank-feature-query.asciidoc]: Re-added with 890b6b3

File: docs/reference/query-dsl/rank-feature-query.asciidoc [hunk 5]
```diff
@@ -4,33 +4,54 @@
...
+
+[[rank-feature-query-index-setup]]
+===== Index setup
+
+To use the `rank_feature` query, your index must include a
+<<rank-feature,`rank_feature`>> or <<rank-features,`rank_features`>> field
+mapping. To see how you can set up an index for the `rank_feature` query, try
+the following example.
+
+. Create a `test` index with the following field mappings:
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/query-dsl/rank-feature-query.asciidoc]: Removed with 356d71f.

File: docs/reference/query-dsl/rank-feature-query.asciidoc [hunk 6]
```diff
@@ -4,33 +4,54 @@
...
 ++++
 
-The `rank_feature` query is a specialized query that only works on
-<<rank-feature,`rank_feature`>> fields and <<rank-features,`rank_features`>> fields.
-Its goal is to boost the score of documents based on the values of numeric
-features. It is typically put in a `should` clause of a
-<<query-dsl-bool-query,`bool`>> query so that its score is added to the score
-of the query.
-
-Compared to using <<query-dsl-function-score-query,`function_score`>> or other
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/query-dsl/rank-feature-query.asciidoc]: Thank you! I relocated this paragraph to the introduction with d225f14.

File: docs/reference/query-dsl/rank-feature-query.asciidoc [hunk 7]
```diff
@@ -4,33 +4,54 @@
...
- - https://en.wikipedia.org/wiki/PageRank[`pagerank`], a measure of the
-   importance of a website,
- - `url_length`, the length of the url, which typically correlates negatively
-   with relevance,
- - `topics`, which associates a list of topics with every document alongside a
-   measure of how well the document is connected to this topic.
-
-Then the example includes an example query that searches for `"2016"` and boosts
-based or `pagerank`, `url_length` and the `sports` topic.
+Boosts the <<query-filter-context,relevance score>> of documents based on the
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/query-dsl/rank-feature-query.asciidoc]: Thanks for the great suggestion. I added a relevance score section to the `query-filter-context` page with ec27f8f. I updated this and other references throughout the doc.

[the remark this answers is not in the document]
Comment (jtibshirani) [reply docs/reference/query-dsl/rank-feature-query.asciidoc]: Looks really nice!

File: docs/reference/query-dsl/rank-feature-query.asciidoc (hunk 4 above)
Comment (jtibshirani) [reply docs/reference/query-dsl/rank-feature-query.asciidoc]: Got it, thanks for the context.

**Review by jtibshirani — APPROVED** [review]

**Pull request merged**